### DISCUSSION TITLE
Consider all imports with the `node:` prefix as server-side only

### DISCUSSION THREAD

**Discussion by sergiodxa** [remix-run/remix#4544] [Proposals]
# Proposal

Consider all imports with the `node:` prefix as server-only packages as if they were importing a `.server.ts` file.

## Background

Node.js docs now suggests using the `node:` prefix when importing built-in packages, Remix could follow this convention to:

1. Ensure Node built-in modules are never sent to the client bundles
2. Let packages with similar names (e.g. buffer) be used client-side
3. Fix errors like #2813

## API/Examples

Importing a Node built-in package:

```ts
import { Buffer } from "node:buffer";
```

Importing the [`buffer`](https://www.npmjs.com/package/buffer) package from npm:

```ts
import { Buffer } from "buffer";
```

## Open Questions

- What should happen with 3P packages importing Node built-in modules without the prefix?

**Comment by pcattori** [comment]
PR: https://github.com/remix-run/remix/pull/5222

Unfortunately, [blocked on upstream polyfills](https://github.com/remix-run/remix/pull/5222#issuecomment-1400719657) for now.

**Comment by pcattori** [comment]
`node:`-prefixed imports are now treated as server-side only (see #5773 )
